demo/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+set(demos
+        "allegro5"
+#        "common"
+#        "d3d11"
+#        "d3d12"
+#        "d3d9"
+#        "gdi"
+#        "gdip"
+#        "glfw_opengl2"
+#        "glfw_opengl3"
+#        "glfw_opengl4"
+#        "sdl2surface_rawfb"
+#        "sdl_opengl2"
+#        "sdl_opengl3"
+#        "sdl_opengles2"
+#        "sdl_renderer"
+#        "sfml_opengl2"
+#        "sfml_opengl3"
+#        "wayland_rawfb"
+#        "x11"
+#        "x11_opengl2"
+#        "x11_opengl3"
+#        "x11_rawfb"
+#        "x11_xft"
+        )
+
+foreach (demo ${demos})
+    add_subdirectory("${demo}")
```

Comment:
The CMakeLists.txt for all these demos are possibly going to be 90% the same, would it make sense to add a cmake function/macro like `add_nuklear_demo(SOURCES xxx LIBS xxx)` for convenience?
(And so that when projects are grouped with something like `set_property(TARGET gdip PROPERTY FOLDER "Nuklear/examples")` this only has to be done in one place?

Reply:
Contributions welcome

Reply:
How do you propose to do this, send a PR to your fork?

Reply:
I'm happy either way